src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/appendoverwrite/factory/entity/PendingObservation.java

```diff
@@ -0,0 +1,198 @@
+     */
+    @Override
+    public List<BrAPIObservation> brapiPost(List<BrAPIObservation> members) throws ApiException, DoesNotExistException {
+        // TODO: move the trait setting out to a higher level
+        // Fetch the program traits
+        List<Trait> traits = ontologyService.getTraitsByProgramId(importContext.getProgram().getId(), true);
+        CaseInsensitiveMap<String, Trait> traitMap = new CaseInsensitiveMap<>();
+        for ( Trait trait: traits) {
+            traitMap.put(trait.getObservationVariableName(),trait);
+        }
+
+        // Set the trait dbId on the observation requests
+        for (BrAPIObservation observation : members) {
+            String observationVariableName = observation.getObservationVariableName();
+            if (observationVariableName != null && traitMap.containsKey(observationVariableName)) {
+                String observationVariableDbId = traitMap.get(observationVariableName).getObservationVariableDbId();
+                observation.setObservationVariableDbId(observationVariableDbId);
+            }
+        }
+
+        // TODO: move this logic out to a higher level
+        // Do not create observations in the BrAPI service if there is no value
+        List<BrAPIObservation> nonBlankMembers = members.stream().filter(obs -> !obs.getValue().isBlank()).collect(Collectors.toList());
+
+        // Create the observations
+        return brAPIObservationDAO.createBrAPIObservations(nonBlankMembers, importContext.getProgram().getId(), importContext.getUpload());
+    }
+
+    /**
+     * Fetch objects required by the workflow from the BrAPI service.
+     *
+     * @return List of fetched entities
+     * @throws ApiException if there is an issue with the API call
+     */
+    @Override
+    public List<BrAPIObservation> brapiRead() throws ApiException {
+        return new ArrayList<>();
+    }
+
+    /**
+     * Commit objects changed by the workflow to the BrAPI service.
+     *
+     * @param members List of entities to be updated
+     * @return List of updated entities
+     * @throws ApiException             if there is an issue with the API call
+     * @throws IllegalArgumentException if method arguments are invalid
+     */
+    @Override
+    public <U> List<U> brapiPut(List<U> members) throws ApiException, IllegalArgumentException {
+        // Check if the input list is of type List<BrAPIObservation>
+        if (experimentUtilities.isInvalidMemberListForClass(members, BrAPIObservation.class)) {
+            return new ArrayList<U>();
+        }
+
+        List<BrAPIObservation> updatedObservations = new ArrayList<>();
+        for (U member : members) {
+            BrAPIObservation observation = (BrAPIObservation) member;
+            Optional.ofNullable(brAPIObservationDAO.updateBrAPIObservation(observation.getObservationDbId(), observation, importContext.getProgram().getId())).ifPresent(updatedObservations::add);
```

Comment:
BI-2093 will impact this area of code so we'll need to make sure it's reflected in the append workflow once it's through review.

Reply:
I created a card for this: [BI-2228](https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/28?search=2228&selectedIssue=BI-2228).